pyproject.toml.in

```diff
@@ -0,0 +1,105 @@
+# this will be turned into pyproject.toml by cmake/ValhallaPyProjectToml.cmake
+# there's no other way to have different "project.name"s
+
+[project]
+name = "@PYVALHALLA_NAME@"
+dynamic = ["version"]
+requires-python = ">=3.9.0"
+authors = [{ name = "Nils Nolde" }, {name = "Kevin Kreiser"}]
+description = "High-level bindings to the Valhalla C++ library"
+readme = "README.md"
+license = "MIT"
+license-files = ["COPYING"]
+
+[project.urls]
+Homepage = "https://github.com/valhalla/valhalla"
+Documentation = "https://valhalla.github.io/valhalla/"
+Repository = "https://github.com/valhalla/valhalla"
+Issues = "https://github.com/valhalla/valhalla/issues"
+Changelog = "https://github.com/valhalla/valhalla/blob/master/CHANGELOG.md"
+
+[project.scripts]
+valhalla_service = "valhalla._scripts:run"
+valhalla_build_tiles = "valhalla._scripts:run"
+valhalla_build_admins = "valhalla._scripts:run"
+valhalla_add_predicted_traffic = "valhalla._scripts:run"
+valhalla_add_landmarks = "valhalla._scripts:run"
+valhalla_ingest_transit = "valhalla._scripts:run"
+valhalla_convert_transit = "valhalla._scripts:run"
+
+[build-system]
+requires = [
+    "scikit-build-core>=0.11.0",
+    "wheel",
+    "setuptools_scm>=6.2",
+    "setuptools_scm_git_archive",
+    "auditwheel",
+    # instead of relying on system tools
+    "cmake",
+    "ninja",
+]
+build-backend = "scikit_build_core.build"
+
+[tool.scikit-build]
+# nanobind’s STABLE_ABI feature targets Python >= 3.12
+wheel.py-api = "cp312"
+# Ask setuptools-scm for the version at build time
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+# creates a valhalla directory in the wheel
+wheel.install-dir = "valhalla"
+# only installs the python cmake COMPONENT
+install.components = ["python"]
+sdist.include = ["pyproject.toml"]
```

Comment:
scikit-build-core by default excludes everything in .gitignore, so it needs the pyproject.toml whitelisted